lib/oli_web/live_session_plugs/authorize_project.ex

```diff
@@ -0,0 +1,31 @@
+defmodule OliWeb.LiveSessionPlugs.AuthorizeProject do
+  use OliWeb, :verified_routes
+  import Phoenix.LiveView, only: [redirect: 2, put_flash: 3]
+
+  alias Oli.Accounts
+  alias Oli.Accounts.Author
+
+  def on_mount(:default, _params, _session, socket) do
+    project = Map.get(socket.assigns, :project)
+    current_author = Map.get(socket.assigns, :current_author)
+
+    case current_author do
+      nil ->
+        {:halt,
+         socket
+         |> put_flash(:error, "You must be logged in to access that project")
+         |> redirect(to: ~p"/workspaces/course_author")}
+
+      %Author{} ->
+        if Accounts.can_access?(current_author, project) &&
+             project.status === :active do
+          {:cont, socket}
```

Comment:
This case should allow admin users to access it as well.

https://github.com/user-attachments/assets/71a891a9-16b2-4bba-8d5f-dd609f1bdb63